StortSpelProjekt/Game/src/Gamefiles/Components/RangeComponent.cpp

```diff
@@ -66,61 +73,77 @@ float component::RangeComponent::GetVelocity()
 	return m_Velocity;
 }
 
-void component::RangeComponent::Attack()
+void component::RangeComponent::SetAttackInterval(float interval)
 {
-	Entity* ent = m_pScene->AddEntity("RangeAttack" + std::to_string(++m_NrOfProjectiles));
-	component::ModelComponent* mc = nullptr;
-	component::TransformComponent* tc = nullptr;
-	component::BoundingBoxComponent* bbc = nullptr;
-	component::ProjectileComponent* pc = nullptr;
-	component::UpgradeComponent* uc = nullptr;
-	component::AccelerationComponent* ac = nullptr;
-
-	mc = ent->AddComponent<component::ModelComponent>();
-	tc = ent->AddComponent<component::TransformComponent>();
-	pc = ent->AddComponent<component::ProjectileComponent>(m_Damage);
-	ac = ent->AddComponent<component::AccelerationComponent>(98.2);
-	uc = ent->AddComponent<component::UpgradeComponent>();
-
-	// Applying all range uppgrades to the new projectile entity "RangeAttack"
-	if (m_pParent->HasComponent<component::UpgradeComponent>())
-	{
-		Player::GetInstance().GetUpgradeManager()->ApplyRangeUpgrades(ent);
-	}
+	m_AttackInterval = interval;
+}
+
+float component::RangeComponent::GetAttackInterval() const
+{
+	return m_AttackInterval;
+}
 
-	// get the pos of parent object and forward of camera 
-	// so we know where to spawn and in which direction
-	float3 ParentPos = m_pParent->GetComponent<component::TransformComponent>()->GetTransform()->GetPositionFloat3();
-	float3 forward = m_pScene->GetMainCamera()->GetDirectionFloat3();
-	float length = forward.length();
-
-	double3 dim = m_pParent->GetComponent<component::ModelComponent>()->GetModelDim();
-
-	// add the forward vector to parent pos 
-	// so the projectile doesn't spawn inside of us
-	float3 pos;
-	pos.x = ParentPos.x + (forward.x / length) * (dim.x / 2.0);
-	pos.y = ParentPos.y + (forward.y / length);
-	pos.z = ParentPos.z + (forward.z / length) * (dim.z / 2.0);
-
-	// initialize the components
-	mc->SetModel(m_pModel);
-	mc->SetDrawFlag(FLAG_DRAW::DRAW_OPAQUE | FLAG_DRAW::GIVE_SHADOW);
-	tc->GetTransform()->SetPosition(pos.x, pos.y, pos.z);
-	tc->GetTransform()->SetMovement(forward.x * m_Velocity, forward.y * m_Velocity, forward.z * m_Velocity);
-	tc->GetTransform()->SetScale(m_Scale);
-	tc->GetTransform()->SetVelocity(m_Velocity); 
-	tc->Update(0.02);
-	bbc = ent->AddComponent<component::BoundingBoxComponent>(F_OBBFlags::COLLISION);
-	bbc->Init();
-	Physics::GetInstance().AddCollisionEntity(ent);
-	if (audioPlay)
+void component::RangeComponent::Attack()
+{
+	if (m_TimeAccumulator > m_AttackInterval)
```

Comment:
minor detail but should be ">="
if we have we have update rate to 0.1 seconds, the true attack interval will be 0.6 seconds instead of the set 0.5